cs-ssh/src/main/java/io/cloudslang/content/ssh/services/impl/SSHServiceImpl.java

```diff
@@ -150,22 +159,32 @@ public CommandResult runShellCommand(
                 currentTime = System.currentTimeMillis();
             }
             boolean timedOut = !channel.isClosed();
-
-            // save the response
             CommandResult result = new CommandResult();
-            result.setStandardOutput(((ByteArrayOutputStream) out).toString(characterSet));
-            result.setStandardError(((ByteArrayOutputStream) err).toString(characterSet));
-
-            channel.disconnect();
-            // The exit status is only available after the channel was closed (more exactly, just before the channel is closed).
-            result.setExitCode(channel.getExitStatus());
-
             if (timedOut) {
                 throw new TimeoutException(String.valueOf(result));
             }
+            byte[] tmp = new byte[1024];
+            while (true) {
+                while (in.available() > 0) {
+                    int i = in.read(tmp, 0, 1024);
+                    if (i < 0) break;
+                    result.setStandardOutput(new String(tmp, 0, i));
```

Comment:
Seems true, what happens when the stream has 1025 characters only one character is present in the message that seems wrong. `in.available` should not be used probably as the loop condition